src/Tests/Cosmos.fs

```diff
@@ -10,44 +10,70 @@ open Microsoft.Rest
-    }
-    test "Correctly serializes to JSON" {
-        let t = arm { add_resource (cosmosDb { name "test" }) }
+            Expect.equal container.UniqueKeys.Count 2 "There should be 2 unique keys."
+            Expect.contains container.UniqueKeys ["/FirstName"] "UniqueKeys should contain /FirstName"
+            Expect.contains container.UniqueKeys ["/LastName"] "UniqueKeys should contain /LastName"
+        }
+        test "DB properties are correctly evaluated" {
+            let db = cosmosDb { name "test" }
+            Expect.equal "[reference(resourceId('Microsoft.DocumentDb/databaseAccounts', 'test-account'), '2020-03-01').documentEndpoint]" (db.Endpoint.Eval()) "Endpoint is incorrect"
+            Expect.equal "[listKeys(resourceId('Microsoft.DocumentDb/databaseAccounts', 'test-account'), providers('Microsoft.DocumentDb','databaseAccounts').apiVersions[0]).primaryMasterKey]" (db.PrimaryKey.Eval()) "Primary Key is incorrect"
+            Expect.equal "[listKeys(resourceId('Microsoft.DocumentDb/databaseAccounts', 'test-account'), providers('Microsoft.DocumentDb','databaseAccounts').apiVersions[0]).secondaryMasterKey]" (db.SecondaryKey.Eval()) "Secondary Key is incorrect"
+            Expect.equal "[listKeys(resourceId('Microsoft.DocumentDb/databaseAccounts', 'test-account'), providers('Microsoft.DocumentDb','databaseAccounts').apiVersions[0]).primaryreadonlyMasterKey]" (db.PrimaryReadonlyKey.Eval()) "Primary Readonly Key is incorrect"
+            Expect.equal "[listKeys(resourceId('Microsoft.DocumentDb/databaseAccounts', 'test-account'), providers('Microsoft.DocumentDb','databaseAccounts').apiVersions[0]).secondaryreadonlyMasterKey]" (db.SecondaryReadonlyKey.Eval()) "Secondary Readonly Key is incorrect"
+            Expect.equal "[listConnectionStrings(resourceId('Microsoft.DocumentDb/databaseAccounts', 'test-account'), providers('Microsoft.DocumentDb','databaseAccounts').apiVersions[0]).connectionStrings[0].connectionString]" (db.PrimaryConnectionString.Eval()) "Primary Connection String is incorrect"
+            Expect.equal "[listConnectionStrings(resourceId('Microsoft.DocumentDb/databaseAccounts', 'test-account'), providers('Microsoft.DocumentDb','databaseAccounts').apiVersions[0]).connectionStrings[1].connectionString]" (db.SecondaryConnectionString.Eval()) "Secondary Connection String is incorrect"
+        }
+        test "Correctly serializes to JSON" {
+            let t = arm { add_resource (cosmosDb { name "test" }) }
 
-        t.Template
-        |> Writer.toJson
-        |> ignore
-    }
-    test "Creates connection string and keys with resource groups" {
-        let conn = CosmosDb.getConnectionString(ResourceId.create("db", "group"), PrimaryConnectionString).Eval()
-        let key = CosmosDb.getKey(ResourceId.create("db", "group"), PrimaryKey, ReadWrite).Eval()
+            t.Template
```

Comment:
Why is everything indented here?

Reply:
I create `invalidAccountNameCases` and `invalidAccountNameCases` inside of `tests` this is why everything is indented. I move those 2 variables outside of `tests` and rollback the formatting changes since they introduce too much noise in here.